app/controllers/ViewChangeNewFileUploadController.scala

```diff
@@ -0,0 +1,172 @@
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  def onPageLoad(srn: Srn, journey: Journey, mode: Mode): Action[AnyContent] =
+    identifyAndRequireData.withFormBundle(srn).async { request =>
+      implicit val dataRequest: DataRequest[AnyContent] = request.underlying
+
+      val formBundleNumber = request.formBundleNumber
+
+      reportDetailsService.getSchemeDetailsItems(formBundleNumber, Pstr(dataRequest.schemeDetails.pstr)).map {
+        schemeDetails =>
+          val preparedForm =
+            dataRequest.userAnswers
+              .fillForm(ViewChangeNewFileUploadPage(srn, journey), ViewChangeNewFileUploadController.form(formProvider))
+
+          Ok(
+            view(
+              preparedForm,
+              ViewChangeNewFileUploadController.viewModel(srn, journey, formBundleNumber, schemeDetails)
+            )
+          )
+      }
+
+    }
+
+  def onSubmit(srn: Srn, journey: Journey, mode: Mode): Action[AnyContent] =
+    identifyAndRequireData.withFormBundle(srn).async { request =>
+      implicit val dataRequest: DataRequest[AnyContent] = request.underlying
+
+      val formBundleNumber = request.formBundleNumber
+
+      reportDetailsService
+        .getSchemeDetailsItems(formBundleNumber, Pstr(dataRequest.schemeDetails.pstr))
+        .flatMap { schemeDetails =>
+          NewFileUploadController
+            .form(formProvider)
+            .bindFromRequest()
+            .fold(
+              formWithErrors =>
+                Future.successful(
+                  BadRequest(
+                    view(
+                      formWithErrors,
+                      ViewChangeNewFileUploadController
+                        .viewModel(srn, journey, formBundleNumber, schemeDetails)
+                    )
+                  )
+                ),
+              value =>
+                for {
+                  updatedAnswers <- Future
+                    .fromTry(dataRequest.userAnswers.set(ViewChangeNewFileUploadPage(srn, journey), value))
+                  _ <- saveService.save(updatedAnswers)
+                  redirectTo <- Future
+                    .successful(
+                      Redirect(navigator.nextPage(ViewChangeNewFileUploadPage(srn, journey), mode, updatedAnswers))
+                    )
+                } yield redirectTo
+            )
+        }
+    }
+}
+
+object ViewChangeNewFileUploadController {
+
+  private val keyBase = "viewChangeNewFileUpload"
+
+  def form(formProvider: UploadNewFileQuestionPageFormProvider): Form[Boolean] =
+    formProvider(s"$keyBase.error.required")
+
+  def viewModel(
+    srn: Srn,
+    journey: Journey,
+    fbNumber: FormBundleNumber,
+    schemeDetailsItems: SchemeDetailsItems
+  ): FormPageViewModel[ViewChangeNewFileQuestionPageViewModel] =
+    journey match {
+      case Journey.InterestInLandOrProperty =>
+        getViewModel(schemeDetailsItems.isLandOrPropertyInterestPopulated, srn, journey, fbNumber)
```

Comment:
might be improved by resolving the scheme details items for a journey with a function and so reducing this block to a single method call?